### PR TITLE
Fix some of the issues found by Coverity

### PR DESCRIPTION
**Pull request by alexey-milovidov**
Changelog category (leave one):
- Build/Testing/Packaging Improvement

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Fix some of the issues found by Coverity. See #19964.

### CODE REVIEW

File: base/common/ReplxxLineReader.cpp [hunk 1]
```diff
@@ -204,7 +206,7 @@ void ReplxxLineReader::openEditor()
         if ((-1 == res || 0 == res) && errno != EINTR)
         {
             rx.print("Cannot write to temporary query file %s: %s\n", filename, errnoToString(errno).c_str());
-            return;
```
Comment (alexey-milovidov) [inline base/common/ReplxxLineReader.cpp:207]: Bug.

File: src/Columns/ColumnDecimal.h [hunk 2]
```diff
@@ -136,7 +136,7 @@ class ColumnDecimal final : public COWHelper<ColumnVectorHelper, ColumnDecimal<T
     Field operator[](size_t n) const override { return DecimalField(data[n], scale); }
     void get(size_t n, Field & res) const override { res = (*this)[n]; }
     bool getBool(size_t n) const override { return bool(data[n].value); }
-    Int64 getInt(size_t n) const override { return Int64(data[n].value * scale); }
```
Comment (alexey-milovidov) [inline src/Columns/ColumnDecimal.h:139]: Potential issue.

File: src/Processors/Formats/Impl/ORCBlockOutputFormat.cpp [hunk 3]
```diff
@@ -133,112 +133,112 @@ ORC_UNIQUE_PTR<orc::Type> ORCBlockOutputFormat::getORCType(const DataTypePtr & t
 
 template <typename NumberType, typename NumberVectorBatch, typename ConvertFunc>
 void ORCBlockOutputFormat::writeNumbers(
-        orc::ColumnVectorBatch * orc_column,
```
Comment (alexey-milovidov) [inline src/Processors/Formats/Impl/ORCBlockOutputFormat.cpp:136]: Bad taste.

File: src/Storages/MergeTree/MergeTreeIndexFullText.cpp [hunk 4]
```diff
@@ -357,7 +357,7 @@ bool MergeTreeConditionFullText::atomFromAST(
             return false;
         }
 
-        if (key_arg_pos == 1 && (func_name != "equals" || func_name != "notEquals"))
```
Comment (alexey-milovidov) [inline src/Storages/MergeTree/MergeTreeIndexFullText.cpp:360]: True bug.

File: base/mysqlxx/Query.cpp [hunk 5]
```diff
@@ -71,16 +71,6 @@ UseQueryResult Query::use()
     return UseQueryResult(res, conn, this);
 }
 
-StoreQueryResult Query::store()
```
Comment (alexey-milovidov) [inline base/mysqlxx/Query.cpp:74]: Simply decided to remove unused code.

File: base/common/ReplxxLineReader.cpp (hunk 1 above)
Comment (amosbird) [reply base/common/ReplxxLineReader.cpp:207]: Hmm, perhaps we need to `close(fd)` and return here, or else it might execute partial query.

**Comment by alexey-milovidov** [comment]
> test_materialize_mysql_database/test.py::test_select_without_columns_8_0[clickhouse_node1]

Flaky test, fixed in master.

File: base/common/ReplxxLineReader.cpp (hunk 1 above)
Comment (alexey-milovidov) [reply base/common/ReplxxLineReader.cpp:207]: That's true.

**Pull request merged**